### DISCUSSION TITLE
Help with 2 codespace

### DISCUSSION THREAD

**Discussion by Yvancg** [community/community#70450] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Hello there,

So I am doing the CS50 course and now reached week 8.

Around week 4, i noticed that the workspace I was working on had a different name than the original one.

It goes like this:
NUMBER[CODESPACES:WORD1-WORD2]
NUMBER[CODESPACES:WORD3-WORD4]
The first one has all my files until week 4, the second one has all my files until now

The change happened (I think) from the moment I switched from the online CS Code version provided by Harvard, after which I used CS Code locally only laptop once I got more confident with coding.

It regularly asks me to restart cs50.dev, which I can then open in CS50 Desktop, but each time by default it will load the first codespace (that stops at week4) and I have to change it manually every time.

Any Idea how I could correct this?
Either sync both codespaces, or select the default codespace that would open, or just move over all the files from the second codespaces to the first...

I am afraid to lose my progress by doing anything wrong and that I would need to resubmit all.

Worse case, I can just suck it up and continue to manually switching until I complete the course.

Thanks in advance for your help.

**Comment by jshorty** [comment, marked as answer]
👋 Hi there!

While I'm unsure exactly what led the CS50 application to create your second codespace, I believe that with CS50 you would typically have write access to the remote GitHub repository that underlies both of your codespaces. What that means is that from either codespace, you should be able to [push your work](https://docs.github.com/en/get-started/using-git/pushing-commits-to-a-remote-repository) to that repository and then [pull and/or merge](https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository) those changes down from the repository into the other codespace.

Regardless of the CS50 application behavior, you can also always manage all your codespaces from https://github.com/codespaces as well, so if you move your work into a single codespace, you should be able to delete the other one.

**Reply by Yvancg** [reply]
I think this may have been triggered by my move from the online application to the desktop application (of VS Code).
In my Github, i can see the newest codespaces listed.
Anyway thanks for replying, that's very kind of you.

**Reply by Yvancg** [reply]
One quick follow-up question:
When i am at the root of the codespaces (NUMBER) on github.com/code50/NUMBER and click on the arrow on the green code button, i see the 2 spaces. Both are shown as active, and besides each of them is an ellipse. When i click on the ... i see a menu. Is it safe to either "stop codespace" or "delete" the one i do not need? Will it impact the other one if i do any of that? 
Apologies for the very newbie question.

**Reply by jshorty** [reply]
It's a bit misleading because of the naming conventions CS50 uses, but `github.com/code50/NUMBER` is a repository with name of `NUMBER`. And nope, stopping or deleting a codespace shouldn't have any direct impacts on the other! The only exception to that would be if you're running services in two active codespaces and forwarding ports so that they're talking to each other, which doesn't sound like your use case. And even in that case, file changes would be persisted when stopping a codespace.

See:
- [about stopping and starting a codespace](https://docs.github.com/en/codespaces/developing-in-codespaces/stopping-and-starting-a-codespace#about-stopping-and-starting-a-codespace)
- [deleting a codespace](https://docs.github.com/en/codespaces/developing-in-codespaces/deleting-a-codespace)

**Reply by Yvancg** [reply]
Thanks again. I will proceed with stopping the unwanted codespace then, and report if anything unwanted happens.
